### PR TITLE
[Design doc] Redesign numeric column clause

### PR DESCRIPTION
**Pull request by sneaxiy**
Remove the `NUMERIC` column clause. We can use the `DENSE` and `SPARSE` to do the same things.

### CODE REVIEW

File: doc/design/numeric_typed_column_clause.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+LABEL label
+COLUMN DENSE(c1, 10), DENSE(c2, 10)
+INTO result_table;
+```
+- For sparse data, the SQL statement would be:
+```sql
+SELECT c1, c2, label FROM train_table
+TO TRAIN DNNRegressor
+LABEL label
+COLUMN SPARSE(c1, 10), SPARSE(c2, 10)
```
Comment (typhoonzero) [inline doc/design/numeric_typed_column_clause.md]: Also add examples of how to embed SPARSE to other columns like `EMBEDDING(SPARSE(...))`?

**Review by typhoonzero — DISMISSED** [review]
LGTM basically

File: doc/design/numeric_column_clause.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        SELECT c1, c2, label FROM train_table
+        TO TRAIN DNNRegressor
+        LABEL label
+        COLUMN NUMERIC(SPARSE(c1, 10)), NUMERIC(SPARSE(c2, 10)) -- 10 is the dense length of the decoded sparse vector
+        INTO result_table;
+        ```
+
+There are some problems in the current design:
+
+- Both `NUMERIC` and `DENSE/SPARSE` have the `length/shape` parameter.
```
Comment (brightcoder01) [inline doc/design/numeric_column_clause.md:87]: Do we support 2 or more dimension? If not, we may not need shape, length is enough

Comment (sneaxiy) [reply doc/design/numeric_column_clause.md:87]: Yes, we should support 2 or more dimensions. For example, some operators like convolution is sensitive to shape but not length.

File: doc/design/numeric_typed_column_clause.md (hunk 1 above)
Comment (sneaxiy) [reply doc/design/numeric_typed_column_clause.md]: Done.

**Review by typhoonzero — APPROVED** [review]
LGTM

**Pull request merged**